cylc/flow/scheduler.py

```diff
@@ -473,6 +473,11 @@ def configure(self):
         if reqmode and not self.config.run_mode(reqmode):
             raise ValueError('this suite requires the %s run mode' % reqmode)
 
+        if not self.is_restart:
+            cp_tz_str = self.config.cfg['cylc']['cycle point time zone']
+            # Will save suite time zone in database:
+            self.options.cp_tz = cp_tz_str
```

Comment:
Should this be set in `self.options` or `self.config` or something else?

Reply:
We shouldn't really be adding extra things to the optparse values object.

Can't we just access this variable directly in `cylc.flow.config` to avoid having to setting anything here.

Reply:
Ah wait, the actual reason I set it in `self.options` was because `Scheduler.load_suiterc()` calls `SuiteConfig()` with the `Scheduler`'s `self.options` as an argument, so I used it to pass `cp_tz` to `SuiteConfig` in order to check if it was set.

Another way would be to add an argument `is_restart` to `SuiteConfig()`

Reply:
https://github.com/cylc/cylc-flow/blob/e9f6e2745b93f9c5fe1117127a12dc30ef65f63f/cylc/flow/scheduler.py#L1137-L1155

Reply:
Yuck, I'll pretend I didn't see it.

Reply:
@oliver-sanders How about adding `cycle_point_tz` as an arg of `SuiteConfig`? ➡️ https://github.com/cylc/cylc-flow/pull/3614/commits/4cc35534f7fb8bacb3a130d0132c9f43579d7978

Reply:
I don't think there is a use case for adding that command line option, and it would be ignored on restart. Happy to leave it.

Reply:
But by replacing `Scheduler.options.cycle_point_tz` with `Scheduler.cycle_point_tz`, and adding the kwarg `cycle_point_tz` to `SuiteConfig()` - doesn't that mean it won't be a command line option any more?

Reply:
I thought the `Scheduler.options` object was for the command line options?

Reply:
`scheduler.options` is for command line options, however, adding a new attribute onto that object does not create a command line option.

Reply:
Should I revert the commit to use `Scheduler.options.cycle_point_tz` then?

Reply:
I think using `scheduler.options` is probably the lesser of two evils, really need a better container for suite params, perhaps in an early version of Cylc8.